app/app_test.go

```diff
@@ -66,3 +88,108 @@ func TestCantoExport(t *testing.T) {
+		evmtypes.StoreKey:         "e5f6f682b247cd9812e05db063d355c3f2661aa12d03cf0b900879a1980d002b",
+		feemarkettypes.StoreKey:   "986b2ae7d92fa552688e4370ae58009e7c2150d41741aae47e748de18bb49f67",
+		govtypes.StoreKey:         "8e160b59f63d310da86e394c68deb47aa2cc47d61afec94b8677d6fb660dae43",
+		ibctransfertypes.StoreKey: "90fdae9ef2124b88045deea846624eed6bc9e5e5422fada847b882fb66d81f17",
+		inflationtypes.StoreKey:   "2f694d43447679f2e058e5629d3305727566e5870dbea6010c574efa8e67ea8e",
+		paramstypes.StoreKey:      "da7007ea1f6f90372c4bf724f30e024a95980353c09b5ed5272de30411604e53",
+		slashingtypes.StoreKey:    "3abe3df91a605993b74446f58c82f17ae1dbebd155cb474c01972befa1e1ca03",
+		stakingtypes.StoreKey:     "48c4ec05af62a8171124c88c2aeca67ca281a71df7047767ba0004d82e80ad84",
+		upgradetypes.StoreKey:     "530a41d3858a8f8fb48ec710e543d633ff0b9ab71500f7c98a13515e53a0f6cd",
+	}
+	// workingHashWithCorrectInitialHeight is the working hash of the IAVL store with correct initial height 4 with given genesis.
+	workingHashWithCorrectInitialHeight := map[string]string{
+		authtypes.StoreKey:        "b513a8bc783341508209bef42f2ca0d97f2e12e3f977e494d30ccd06ba2049f3",
+		banktypes.StoreKey:        "cf0406a0e743fd4297d67b80dc245c81c466d722c8f2415c97b892d1da592c19",
+		capabilitytypes.StoreKey:  "e9261548b1c687638721f75920e1c8e3f4f52cbd7ab3aeddc6c626cd8abc8718",
+		coinswaptypes.StoreKey:    "57094d1ec4776eab36ae55145557a8679dca01529feed5e83bb1753a8849ed28",
+		consensustypes.StoreKey:   "2573460b2ef3a08c825bdfb485e51680038530f70c45f0b723167ae09599761c",
+		crisistypes.StoreKey:      "7c169f2c8fa4c6f64a61154e4fb3ffb3d74893a8bf6efdf50fc6e06d92979ecb",
+		distrtypes.StoreKey:       "65cfb5c307b023ed80255b3ffc14e08b39cc00ecd7b710ff8b5bc96d1efdbda6",
+		epochstypes.StoreKey:      "a26294cd8405c0e3cabc508c90d34317bfe547c90fd4be22223d27aecd819211",
+		evmtypes.StoreKey:         "f2dcba601044394f83455be931d2ad78a08ede45873b16d8884d5f650db42f99",
+		feemarkettypes.StoreKey:   "b8a66cce8e7809f521db9fdd71bfeb980966f1b74ef252bda65804d8b89da7de",
+		govtypes.StoreKey:         "033e4a113195025eb54ecdeabffec5fd20605eae08da125d237768f1f3387616",
+		ibctransfertypes.StoreKey: "3ffd548eb86288efc51964649e36dc710f591c3d60d6f9c1b42f2a4d17870904",
+		inflationtypes.StoreKey:   "b85bc597af9eb62c42e06b0f158bde591975585bab384e9666f89f80001b3d01",
+		paramstypes.StoreKey:      "689664bc7d4f9aa8abeeb6ba13a556467e149d56e7bead89e8b574bdf8e0fe7f",
+		slashingtypes.StoreKey:    "9da3ff2ded57e30dfea0371278d9043bea9f579421beb45b58ec7240e1b4f27a",
+		stakingtypes.StoreKey:     "17b36186121d21b713a667c6cd534562bbc3095974ec866cb906cad7c2fa31e1",
+		upgradetypes.StoreKey:     "9677219870ca98ba9868589ccdcd97411d9b82abc6a7aa1910016457b4730a93",
+	}
+
+	matchAny := func(key string) bool {
+		for _, dk := range deterministicKeys {
+			if dk == key {
+				return true
+			}
+		}
+		return false
+	}
+
+	for _, key := range storeKeys {
+		if key != nil && matchAny(key.Name()) {
```

Comment:
```suggestion
		if matchAny(key.Name()) {
```
`key` is not pointer, so i think you no need to nil check

Reply:
@dudong2 
key can be nil. I ran the following code and the result was 0x19.
```go
	counter := 0
	for _, key := range storeKeys {
		if key == nil {
			counter++
		}
		if key != nil && matchAny(key.Name()) {
			kvstore := app.CommitMultiStore().GetCommitKVStore(key)
			require.Equal(t, storetypes.StoreTypeIAVL, kvstore.GetStoreType())
			iavlStore, ok := kvstore.(*iavl.Store)
			require.True(t, ok)
			workingHash := hex.EncodeToString(iavlStore.WorkingHash())
			require.NotEqual(t, workingHashWithZeroInitialHeight[key.Name()], workingHash)
			require.Equal(t, workingHashWithCorrectInitialHeight[key.Name()], workingHash)
		}
	}
	fmt.Println(counter)

```